packages/sanity/src/core/form/studio/tree-editing/utils/build-tree-editing-state/buildArrayState.ts

```diff
@@ -122,6 +123,19 @@ export function buildArrayState(props: BuildArrayState): TreeEditingState {
           schemaType: childField as ObjectSchemaType,
           value: childValue,
         })
+
+        return
+      }
+
+      // If `openPath` points to an array field within a portable text field,
+      // set `relativePath` to the parent of the portable text field.
+      // This ensures that the tree editing dialog opens at the parent level
+      // of the portable text field.
+      // Portable text fields manage their own dialogs, so we open the tree editing
+      // dialog for the parent item and let the portable text field handle its
+      // dialogs via `openPath`.
+      if (isPortableText && toString(openPath).startsWith(toString(childPath))) {
```

Comment:
Small nit: think you can use [`startsWith`](https://github.com/sanity-io/sanity/blob/corel-35/packages/@sanity/util/src/paths.ts#L126) from the Path Utils :)